AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3658,6 +3658,167 @@ class CheckResult {
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="Resource group name")]
+        [string]$ResourceGroupName,
+
+        [Parameter(Mandatory=$False, Position=2, HelpMessage="Filter")]
+        [string]$Filter,
+
+        [Parameter(Mandatory=$False, Position=3, HelpMessage="Optional parameter for filter 'CheckSidHasAadUser' and 'CheckUserFileAccess'. The user name to check.")]
+        [string]$UserName,
+
+        [Parameter(Mandatory=$False, Position=4, HelpMessage="Optional parameter for filter 'CheckSidHasAadUser', 'CheckUserFileAccess' and 'CheckAadUserHasSid'. The domain name to look up the user.")]
+        [string]$Domain,
+
+        [Parameter(Mandatory=$False, Position=5, HelpMessage="Required parameter for filter 'CheckAadUserHasSid'. The Azure object ID or user principal name to check.")]
+        [string]$ObjectId,
+
+        [Parameter(Mandatory=$False, Position=6, HelpMessage="Required parameter for filter 'CheckUserFileAccess'. The SMB file path on the Azure file share mounted locally using storage account key.")]
+        [string]$FilePath
+    )
+
+    process
+    {
+        $checksExecuted = 0;
+        $filterIsPresent = ![string]::IsNullOrEmpty($Filter);
+        $checks = @{
+            "CheckPort445Connectivity" = [CheckResult]::new("CheckPort445Connectivity");
+            "CheckAADConnectivity" = [CheckResult]::new("CheckAADConnectivity");
+        }
+        #
+        # Port 445 check 
+        #
+        
+        if (!$filterIsPresent -or $Filter -match "CheckPort445Connectivity")
+        {
+            try {
+                $checksExecuted += 1;
+                Write-Verbose "CheckPort445Connectivity - START"
+
+                Test-Port445Connectivity -StorageAccountName $StorageAccountName `
+                    -ResourceGroupName $ResourceGroupName -ErrorAction Stop
+
+                $checks["CheckPort445Connectivity"].Result = "Passed"
+                Write-Verbose "CheckPort445Connectivity - SUCCESS"
+            } catch {
+                $checks["CheckPort445Connectivity"].Result = "Failed"
+                $checks["CheckPort445Connectivity"].Issue = $_
+                Write-Error "CheckPort445Connectivity - FAILED"
+                Write-Error $_
+            }
+        }
+        if (!$filterIsPresent -or $Filter -match "CheckAADConnectivity")
+        {
+            try {
+                $checksExecuted += 1;
+                Write-Verbose "CheckAADConnectivity - START"
+                $TenantID = Get-AzContext
+                $Tenant = $TenantID.Tenant
+                $Response = Invoke-WebRequest -Method POST https://login.microsoftonline.com/$Tenant/kerberos
+                if ($Response.StatusCode == 200)
+                {
+                    $checks["CheckAADConnectivity"].Result = "Passed"
+                    Write-Verbose "CheckAADConnectivity - SUCCESS"
+                }

```

Comment:
What if the response status code isn't 200? We need to have an `else` here that writes a failure to the check result. I think anything else than 200 is highly unexpected, but you never know (maybe they're on some public wifi and they're getting a HTTP 301 redirect, or something like that. Or maybe there's downtime in AAD, and they're receiving a 500 error. Or maybe their tenant doesn't exist and they're getting a 404?)

We don't need a case for each of these, we can just log the status code and that the check failed.